core/src/main/java/com/salesforce/slds/validation/validators/impl/recommendation/MobileFriendlyValidator.java

```diff
@@ -0,0 +1,120 @@
+import com.salesforce.slds.shared.models.recommendation.Action;
+import com.salesforce.slds.shared.models.recommendation.ActionType;
+import com.salesforce.slds.shared.models.recommendation.Item;
+import com.salesforce.slds.shared.models.recommendation.Recommendation;
+import com.salesforce.slds.shared.utils.ResourceUtilities;
+import com.salesforce.slds.validation.validators.interfaces.RecommendationValidator;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Component
+@Lazy
+public class MobileFriendlyValidator implements RecommendationValidator, InitializingBean {
+    Map<String, List<String>> componentsRegistry;
+
+    @Override
+    public List<Recommendation> matches(Entry entry, Bundle bundle, Context context) {
+        if (entry.getEntityType() != Entry.EntityType.LWC) {
+            return Lists.newArrayList();
+        }
+
+        return entry.getInputs().stream().filter(input -> input.asElement() != null)
+                .map(Input::asElement)
+                .filter(htmlElement -> !isMobileFriendly(htmlElement))
+                .map(htmlElement -> {
+                    Recommendation.RecommendationBuilder builder = Recommendation.builder()
+                            .input(htmlElement);
+
+                    String tag = htmlElement.getContent().tagName();
+
+                    Action action = Action.builder()
+                            .value(htmlElement.getContent().outerHtml()).range(htmlElement.getRange())
+                            .description(tag
+                                    + " is known to have issues on mobile devices. Consider using a replacement " +
+                                    "or create a custom component to use instead.")
+                            .name(tag)
+                            .actionType(ActionType.NONE)
+                            .build();
+
+                    Item item = new Item(tag, action);
+                    builder.items(Sets.newHashSet(item));
+
+                    return builder.build();
+                }).collect(Collectors.toList());
+    }
+
+    private boolean isMobileFriendly(HTMLElement input) {
+        String tag = input.getContent().tagName();
+        String convertedTag = convertToNamespaceCamelCaseSyntax(tag);
+
+        List<String> compatibilitySettings = this.componentsRegistry.get(convertedTag);
+        if (compatibilitySettings != null) {
+            boolean compatible = compatibilitySettings.stream().anyMatch(s ->
+                    s.equalsIgnoreCase("Standard") || s.equalsIgnoreCase("Mobile"));
+            return compatible;
+        }
+        return true;
+    }
+
+    private String convertToNamespaceCamelCaseSyntax(String tag) {
```

Comment:
nit: doesn't have to be part of this PR but this function seems to be perfect candidate to be moved to a util class and shared for other classes to use if they need to.

Reply:
As part of that exercise, it would also be helpful to doc it, with examples of input and output.